api/v1alpha1/basic_auth_types.go

```diff
@@ -24,4 +27,10 @@ type BasicAuth struct {
 	//
 	// Note: The secret must be in the same namespace as the SecurityPolicy.
 	Users gwapiv1b1.SecretObjectReference `json:"users"`
+
+	// The name of the HTTP header that will be used to forward the username to the upstream server.
+	// Please note that the header name is a global setting for all the routes in a Gateway.
```

Comment:
`UserNameToHeader` can also be per route, but will need some upstream work.

Reply:
imo we should wait until per route support is added, else a per route setting will enable this at the global level

Reply:
OK, let's wait for the upstream.